### PR TITLE
API-69: Apply permissions on products in API

### PR DESCRIPTION
**Pull request by skeleton**
[//]: <> (<3 Thanks for taking the time to contribute! You're awesome! <3)

- Remove some clear of the unit of work, they are useless now.
- Refactor a little bit integration tests for products to reuse them for EE
- Create a new role for integration tests. Before role `redactor` was not allowed to access to the API. It's problematic because with permissions, we always talk about mary who is a redactor. As permissions are a enough complicated with good roles, I allow now Mary to see the API, and created a `trainee` role who cannot see the API. So welcome to Kevin Michel the Trainee :)
 
[//]: <> (If you've never contributed to this repository before, please read https://github.com/akeneo/pim-community-dev/blob/master/.github/CONTRIBUTING.md)

**Description (for Contributor and Core Developer)**

[//]: <> (What does this Pull Request do? reference the related issue?)

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added Behats                      | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: features/Context/WebUser.php [hunk 1]
```diff
@@ -1779,7 +1779,6 @@ public function productShouldBeEnabled(Product $product)
     public function theFamilyOfProductShouldBe($sku, $expectedFamily = '')
     {
         $this->spin(function () use ($sku, $expectedFamily) {
-            $this->clearUOW();
```
Comment (solivier) [inline features/Context/WebUser.php]: :+1:

File: src/Pim/Bundle/ApiBundle/tests/integration/Security/AttributeAuthorizationIntegration.php [hunk 2]
```diff
@@ -11,7 +11,7 @@ class AttributeAuthorizationIntegration extends ApiTestCase
 {
     public function testOverallAccessDenied()
     {
-        $client = $this->createAuthenticatedClient([], [], null, null, 'mary', 'mary');
+        $client = $this->createAuthenticatedClient([], [], null, null, 'kevin', 'kevin');
```
Comment (solivier) [inline src/Pim/Bundle/ApiBundle/tests/integration/Security/AttributeAuthorizationIntegration.php]: :-1:

Comment (skeleton) [inline src/Pim/Bundle/ApiBundle/tests/integration/Security/AttributeAuthorizationIntegration.php]: you prefer Oliver ?

File: src/Pim/Component/User/Authentication/SuperAdminTokenStorage.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+use Akeneo\Component\StorageUtils\Factory\SimpleFactoryInterface;
+use Doctrine\Common\Persistence\ObjectRepository;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authentication\Token\UsernamePasswordToken;
+
+/**
+ * Generate a user on the fly with all permissions and create a token in storage with this user
+ * to be able to use the PIM without force authentication (from CLI for instance).
```
Comment (ahocquard) [inline src/Pim/Component/User/Authentication/SuperAdminTokenStorage.php]: without forcing

**Comment by jjanvier** [comment]
ok, about the factory I understood by reading the EE PR

**Pull request closed without merging**